cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -503,38 +503,77 @@ func TestHTTPProxyCIDRCheck(t *testing.T) {
 	}
 
 	// Save current content of *_proxy and *_PROXY variables.
-	savedEnv := resetProxyEnv()
+	savedEnv := resetProxyEnv(t)
 	defer restoreEnv(savedEnv)
-	t.Log("Saved environment: ", savedEnv)
-
-	os.Setenv("HTTP_PROXY", "http://proxy.example.com:3128")
-	os.Setenv("NO_PROXY", "example.com,10.0.0.0/8,2001:db8::/48")
 
-	// Check if we can reliably execute tests:
-	// ProxyFromEnvironment caches the *_proxy environment variables and
-	// if ProxyFromEnvironment already executed before our test with empty
-	// HTTP_PROXY it will make these tests return false positive failures
-	req, err := http.NewRequest("GET", "http://host.fake.tld/", nil)
-	if err != nil {
-		t.Fatalf("unexpected err: %v", err)
-	}
-	proxy, err := http.ProxyFromEnvironment(req)
-	if err != nil {
-		t.Fatalf("unexpected err: %v", err)
+	for _, rt := range tests {
+		warning, _ := rt.check.Check()
+		if (warning != nil) != rt.expectWarnings {
+			t.Errorf(
+				"failed HTTPProxyCIDRCheck:\n\texpected: %t\n\t  actual: %t (CIDR:%s). Warnings: %v",
+				rt.expectWarnings,
+				(warning != nil),
+				rt.check.CIDR,
+				warning,
+			)
+		}
 	}
-	if proxy == nil {
-		t.Skip("test skipped as ProxyFromEnvironment already initialized in environment without defined HTTP proxy")
+}
+
+func TestHTTPProxyCheck(t *testing.T) {
+	var tests = []struct {
+		check          HTTPProxyCheck
+		expectWarnings bool
+	}{
+		{
+			check: HTTPProxyCheck{
+				Proto: "https",
+				Host:  "127.0.0.1",
+			}, // Loopback addresses never should produce proxy warnings
+			expectWarnings: false,
+		},
+		{
+			check: HTTPProxyCheck{
+				Proto: "https",
+				Host:  "10.96.0.1",
+			}, // Expected to be accessed directly, we set NO_PROXY to 10.0.0.0/8
+			expectWarnings: false,
+		},
+		{
+			check: HTTPProxyCheck{
+				Proto: "https",
+				Host:  "192.168.0.1",
+			}, // Expected to go via proxy as this range is not listed in NO_PROXY
+			expectWarnings: true,
+		},
+		{
+			check: HTTPProxyCheck{
+				Proto: "https",
+				Host:  "2001:db8::1:15",
+			}, // Expected to be accessed directly, part of 2001:db8::/48 in NO_PROXY
+			expectWarnings: false,
+		},
+		{
+			check: HTTPProxyCheck{
+				Proto: "https",
+				Host:  "2001:db8:1::1:15",
+			}, // Expected to go via proxy, range is not in 2001:db8::/48
+			expectWarnings: true,
+		},
 	}
-	t.Log("http.ProxyFromEnvironment is usable, continue executing test")
+
+	// Save current content of *_proxy and *_PROXY variables.
+	savedEnv := resetProxyEnv(t)
+	defer restoreEnv(savedEnv)
 
 	for _, rt := range tests {
 		warning, _ := rt.check.Check()
 		if (warning != nil) != rt.expectWarnings {
 			t.Errorf(
-				"failed HTTPProxyCIDRCheck:\n\texpected: %t\n\t  actual: %t (CIDR:%s). Warnings: %v",
+				"failed HTTPProxyCheck:\n\texpected: %t\n\t  actual: %t (CIDR:%s). Warnings: %v",
```

Comment:
in this line then "CIDR:%s" should become "Host:%s"

Reply:
nit: if you want, you could make the error message more descriptive and code more self-documenting, by added a description field to the test structure and use the comments (above) as the description text.  This way, it is easier to tell which test failed, versus comparing the Host IP to the structs above.

Reply:
Will add both these suggestions.